internal/elasticsearch/pipeline/resource.go

```diff
@@ -0,0 +1,49 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipeline
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+// Resource represents a pipeline resource loaded from a file
+type Resource struct {
```

Comment:
I'm thinking loud now, as the _Resource_ may be too vague.

An alternative approach would be:
- create the `ingest` package in the `internal/elasticsearch` directory
- add `pipelines.go` 
- create the structure _Pipeline_.

or just move this file (with unit test) to the `internal/elasticsearch` directory, and rename to _Pipeline_. WDYT?

Reply:
done